eliot/_message.py

```diff
@@ -106,16 +124,57 @@ def write(self, logger=None, action=None):
         """
         if logger is None:
             logger = _output._DEFAULT_LOGGER
-        if action is None:
-            action = currentAction()
-        if action is None:
-            action = _defaultAction
-        contents = self._contents.copy()
-        contents["timestamp"] = self._timestamp()
-        if action is not None:
-            contents["task_uuid"] = action._identification["task_uuid"]
-            contents["task_level"] = action._nextTaskLevel().level
-        logger.write(contents, self._serializer)
+        logged = self._freeze(action=action)
+        logger.write(logged.as_dict(), self._serializer)
```

Comment:
As you say, returning `WrittenMessage` from here, so I've removed it. (YAGNI and all that).

Original motivation was to make sure that there was some public API from which it was possible to get a `WrittenMessage`, and also a partially left-over refactoring